src/Catel.Core/Catel.Core.Shared/Data/ModelBase.editableobject.cs

```diff
@@ -308,25 +308,32 @@ void IEditableObject.CancelEdit()
         /// </summary>
         void IEditableObject.EndEdit()
         {
-            if (_backup == null)
+            try
             {
-                Log.Debug("IEditableObject is not in edit state");
-                return;
-            }
+                if (_backup == null)
+                {
+                    Log.Debug("IEditableObject is not in edit state");
+                    return;
+                }
 
-            var eventArgs = new EndEditEventArgs(this);
-            _endEditingEvent.SafeInvoke(this, eventArgs);
-            OnEndEdit(eventArgs);
+                var eventArgs = new EndEditEventArgs(this);
+                _endEditingEvent.SafeInvoke(this, eventArgs);
+                OnEndEdit(eventArgs);
 
-            if (eventArgs.Cancel)
-            {
-                Log.Info("IEditableObject.EndEdit is canceled by the event args");
-                return;
-            }
+                if (eventArgs.Cancel)
+                {
+                    Log.Info("IEditableObject.EndEdit is canceled by the event args");
+                    return;
+                }
 
-            Log.Debug("IEditableObject.EndEdit");
+                Log.Debug("IEditableObject.EndEdit");
 
-            _backup = null;
+                _backup = null;
+            }
+            finally
+            {
+                IsDirty = false;
```

Comment:
We don't know if IsDirty should be reset to false. When we create the backup, we should store the IsDirty and restore that value.